src/YesSql.Core/Session.cs

```diff
@@ -1233,11 +1233,11 @@ private async Task ReduceAsync()
             }
         }
 
-        private async Task<ReduceIndex> ReduceForAsync(IndexDescriptor descriptor, object currentKey)
+        private async Task<ReduceIndex> ReduceForAsync(IndexDescriptor descriptor, object currentKey, string collection)
         {
             await DemandAsync();
 
-            var name = _tablePrefix + descriptor.IndexType.Name;
+            var name = _tablePrefix + _store.Configuration.TableNameConvention.GetIndexTable(descriptor.IndexType, collection);
```

Comment:
so the only real change here is to pass the current collection name into the reduce query.

Everything else in the pr, is just reducing `Collection1` down to `Col1` so MySql doesn't have a fit about index name lengths

Reply:
@deanmarcussen 

We already reduced index names but it was for `PostgreSql` that truncates too long names, but i thought it was okay for `MySql` for which we had some problems but for index row sizes, see https://github.com/OrchardCMS/OrchardCore/issues/8649#issue-813113977

Good to know if index name lengths is also a problem with `MySql`, but i would just want to confirm it

Reply:
Both MySql and Postgres are limited to 64 characters (table names, and indexes - basically everything).

The errors related to them just show up in different ways. (I think Postgres truncates, MySql throws, but it seems inconsistent).

If you give MySql no index name, it actually generates a unique one, so maybe something to look into.
But then we lose track of the index name. And am not sure if Postgres has the same feature

Reply:
Okay thanks

And i think that Postgres truncates before the 63 chars, that would explain why we had an issue on Postgres with 2 indexes starting with a same and quite long string, but not on MySql.

So yes, better to try it concretely, thanks for doing that